src/evolution_scene.c

```diff
@@ -878,350 +937,403 @@ static void Task_EvolutionScene(u8 taskID)
+    T_EVOSTATE_SPARKLE_ARC,
+    T_EVOSTATE_CYCLE_MON_SPRITE,
+    T_EVOSTATE_WAIT_CYCLE_MON_SPRITE,
+    T_EVOSTATE_SPARKLE_CIRCLE,
+    T_EVOSTATE_SPARKLE_SPRAY,
+    T_EVOSTATE_EVO_SOUND,
+    T_EVOSTATE_EVO_MON_ANIM,
+    T_EVOSTATE_SET_MON_EVOLVED,
+    T_EVOSTATE_TRY_LEARN_MOVE,
+    T_EVOSTATE_END,
+    T_EVOSTATE_CANCEL,
+    T_EVOSTATE_CANCEL_MON_ANIM,
+    T_EVOSTATE_CANCEL_MSG,
+    T_EVOSTATE_LEARNED_MOVE,
+    T_EVOSTATE_TRY_LEARN_ANOTHER_MOVE,
+    T_EVOSTATE_REPLACE_MOVE,
+};
+
+// States for the switch in T_EVOSTATE_REPLACE_MOVE
+enum {
+    T_MVSTATE_INTRO_MSG_1,
+    T_MVSTATE_INTRO_MSG_2,
+    T_MVSTATE_INTRO_MSG_3,
+    T_MVSTATE_PRINT_YES_NO,
+    T_MVSTATE_HANDLE_YES_NO,
+    T_MVSTATE_SHOW_MOVE_SELECT,
+    T_MVSTATE_HANDLE_MOVE_SELECT,
+    T_MVSTATE_FORGET_MSG,
+    T_MVSTATE_LEARNED_MOVE,
+    T_MVSTATE_ASK_CANCEL,
+    T_MVSTATE_CANCEL,
+    T_MVSTATE_RETRY_AFTER_HM,
+};
+
+// Compare to Task_EvolutionScene, very similar
+static void Task_TradeEvolutionScene(u8 taskId)
 {
     u32 var = 0;
-    struct Pokemon* mon = &gPlayerParty[gTasks[taskID].tPartyID];
+    struct Pokemon* mon = &gPlayerParty[gTasks[taskId].tPartyId];
 
-    switch (gTasks[taskID].tState)
+    switch (gTasks[taskId].tState)
     {
-    case 0:
+    case T_EVOSTATE_INTRO_MSG:
         StringExpandPlaceholders(gStringVar4, gText_PkmnIsEvolving);
         DrawTextOnTradeWindow(0, gStringVar4, 1);
-        gTasks[taskID].tState++;
+        gTasks[taskId].tState++;
         break;
-    case 1:
+    case T_EVOSTATE_INTRO_CRY:
         if (!IsTextPrinterActive(0))
         {
-            PlayCry1(gTasks[taskID].tPreEvoSpecies, 0);
-            gTasks[taskID].tState++;
+            PlayCry1(gTasks[taskId].tPreEvoSpecies, 0);
+            gTasks[taskId].tState++;
         }
         break;
-    case 2:
+    case T_EVOSTATE_INTRO_SOUND:
         if (IsCryFinished())
         {
             m4aSongNumStop(MUS_EVOLUTION);
             PlaySE(MUS_EVOLUTION_INTRO);
-            gTasks[taskID].tState++;
+            gTasks[taskId].tState++;
         }
         break;
-    case 3:
+    case T_EVOSTATE_START_MUSIC:
         if (!IsSEPlaying())
         {
             PlayBGM(MUS_EVOLUTION);
-            gTasks[taskID].tState++;
+            gTasks[taskId].tState++;
             BeginNormalPaletteFade(0x1C, 4, 0, 0x10, RGB_BLACK);
         }
         break;
-    case 4:
+    case T_EVOSTATE_START_BG_AND_SPARKLE_SPIRAL:
         if (!gPaletteFade.active)
         {
-            InitMovingBackgroundTask(TRUE);
-            var = gSprites[sEvoStructPtr->preEvoSpriteID].oam.paletteNum + 16;
-            sEvoGraphicsTaskID = LaunchTask_PreEvoSparklesSet1(var);
-            gTasks[taskID].tState++;
+            StartBgAnimation(TRUE);
+            var = gSprites[sEvoStructPtr->preEvoSpriteId].oam.paletteNum + 16;
+            sEvoGraphicsTaskId = EvolutionSparkles_SpiralUpward(var);
+            gTasks[taskId].tState++;
             SetGpuReg(REG_OFFSET_BG3CNT, 0x603);
```

Comment:
should be using bg cnt constants

Reply:
There's lots of missing constant usage. This has been merged, if you'd like to add the constant usage please open a PR